File: plugins/umeng/ios/UmengPushModule.swift
import Foundation
import React

/// Stub wrapper that mirrors the UMeng Push API the Vue layer expects.
///
/// Currently each method resolves with `nil` so the app compiles without the UMeng SDK.
/// To enable real push:
///   1. Add `pod 'UMCommon'`, `pod 'UMDevice'`, `pod 'UMPush'` via plugins/withUmengPush.js.
///   2. Import `UMCommon` + `UMPush` below.
///   3. Call `UMConfigure.initWithAppkey(...)` and `UMessage.start(...)`.
///   4. Register APNs via `UMessage.register(forRemoteNotifications:withAppKey:)`.
@objc(UmengPushModule)
class UmengPushModule: NSObject {

  @objc static func requiresMainQueueSetup() -> Bool { false }

  @objc func initPush(
    _ resolve: @escaping RCTPromiseResolveBlock,
    rejecter reject: @escaping RCTPromiseRejectBlock
  ) {
    // TODO(UMeng):
    // UMConfigure.initWithAppkey(appKey, channel: "expo")
    // UMConfigure.setLogEnabled(false)
    // let entity = UMessageRegisterEntity()
    // entity.types = Int(UMessageAuthorizationOptionBadge.rawValue
    //   | UMessageAuthorizationOptionSound.rawValue
    //   | UMessageAuthorizationOptionAlert.rawValue)
    // UNUserNotificationCenter.current().delegate = self
    // UMessage.register(forRemoteNotifications: entity, withAppKey: appKey) { ... }
    NSLog("[UmengPush] initPush stub — wire UMeng SDK to enable push")
    resolve(nil)
  }

  @objc func setAlias(
    _ alias: String,
    type: String,
    resolver resolve: @escaping RCTPromiseResolveBlock,
    rejecter reject: @escaping RCTPromiseRejectBlock
  ) {
    // TODO(UMeng): UMessage.addAlias(alias, type: type) { ... }
    NSLog("[UmengPush] setAlias stub: %@=%@", type, alias)
    resolve(nil)
  }

  @objc func removeAlias(
    _ alias: String,
    type: String,
    resolver resolve: @escaping RCTPromiseResolveBlock,
    rejecter reject: @escaping RCTPromiseRejectBlock
  ) {
    // TODO(UMeng): UMessage.removeAlias(alias, type: type) { ... }
    NSLog("[UmengPush] removeAlias stub: %@=%@", type, alias)
    resolve(nil)
  }

  @objc func getDeviceToken(
    _ resolve: @escaping RCTPromiseResolveBlock,
    rejecter reject: @escaping RCTPromiseRejectBlock
  ) {
    // TODO(UMeng): resolve(UMessage.deviceToken())
    resolve(nil)
  }
}

File: android/app/src/main/java/com/dootask/umeng/UmengPushModule.kt
package com.dootask.umeng

import android.util.Log
import com.facebook.react.bridge.Promise
import com.facebook.react.bridge.ReactApplicationContext
import com.facebook.react.bridge.ReactContextBaseJavaModule
import com.facebook.react.bridge.ReactMethod

/**
 * Stub wrapper that mirrors the UMeng Push API surface the Vue layer expects.
 *
 * The methods currently log and resolve with null so the app compiles without the UMeng SDK.
 * To enable real push:
 *   1. Add UMeng Maven repo + dependencies via `plugins/withUmengPush.js` (options section).
 *   2. Uncomment the UMeng SDK imports + calls below.
 *   3. Provide UMENG_APPKEY / UMENG_MESSAGE_SECRET / channel via plugin options.
 *   4. Call UMConfigure.init before PushAgent.register.
 */
class UmengPushModule(reactContext: ReactApplicationContext) :
    ReactContextBaseJavaModule(reactContext) {

    override fun getName() = "UmengPush"

    @ReactMethod
    fun initPush(promise: Promise) {
        // TODO(UMeng): move UMConfigure.init and PushAgent.register here.
        // val appKey = BuildConfig.UMENG_APPKEY
        // val messageSecret = BuildConfig.UMENG_MESSAGE_SECRET
        // UMConfigure.init(
        //     reactApplicationContext,
        //     appKey,
        //     "expo",
        //     UMConfigure.DEVICE_TYPE_PHONE,
        //     messageSecret,
        // )
        // PushAgent.getInstance(reactApplicationContext).register(object : UPushRegisterCallback {
        //     override fun onSuccess(deviceToken: String) { promise.resolve(deviceToken) }
        //     override fun onFailure(errCode: String, errDesc: String) {
        //         promise.reject(errCode, errDesc)
        //     }
        // })
        Log.w(TAG, "initPush stub — wire UMeng SDK to enable push")
        promise.resolve(null)
    }

    @ReactMethod
    fun setAlias(alias: String, type: String, promise: Promise) {
        // TODO(UMeng):
        // PushAgent.getInstance(reactApplicationContext)
        //     .setAlias(alias, type, UPushAliasCallback { isSuccess, message ->
        //         if (isSuccess) promise.resolve(null) else promise.reject("setAlias", message)
        //     })
        Log.w(TAG, "setAlias stub: $type=$alias")
        promise.resolve(null)
    }

    @ReactMethod
    fun removeAlias(alias: String, type: String, promise: Promise) {
        // TODO(UMeng):
        // PushAgent.getInstance(reactApplicationContext)
        //     .deleteAlias(alias, type, UPushAliasCallback { isSuccess, message ->
        //         if (isSuccess) promise.resolve(null) else promise.reject("removeAlias", message)
        //     })
        Log.w(TAG, "removeAlias stub: $type=$alias")
        promise.resolve(null)
    }

    @ReactMethod
    fun getDeviceToken(promise: Promise) {
        // TODO(UMeng):
        // promise.resolve(PushAgent.getInstance(reactApplicationContext).registrationId)
        promise.resolve(null)
    }

    companion object {
        private const val TAG = "UmengPush"
    }
}
